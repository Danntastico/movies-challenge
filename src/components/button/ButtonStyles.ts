import  styled from 'styled-components';
import { Button } from 'react-bootstrap'

export const ButtonComp = styled(Button)`
    background-color: ${({customColor}) => customColor ? customColor : "transparent"};
    border: transparent;

    :active  {
        
        border-color: transparent;
        background-color: black;
        color: white;
        
    }

    &:hover {
        background-color: ${({customHoverColor}) => customHoverColor ? customHoverColor : "transparent"};
    }

    @media screen and (max-width: 613px) {
        padding: 0 1px 0 1px;
    }
`

