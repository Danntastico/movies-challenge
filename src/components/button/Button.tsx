import React, { FunctionComponent } from 'react';
import { ButtonComp } from './ButtonStyles';
interface ButtonProps {
    onClick: () => void;
    children: any;
    customColor?:string;
    customHoverColor?: string;
    customTextColor?:string;
    customTextHoverColor?: string;
}

export const Button = ({ 
        onClick, 
        children}: ButtonProps) => {

    return (
        <>
            <ButtonComp 
                onClick={onClick}
            >
                { children }
            </ButtonComp>
        </>
    )
}
