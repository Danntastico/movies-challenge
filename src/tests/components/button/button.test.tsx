import React from 'react';
import { mount } from 'enzyme';
import {Button} from '../../../components/button/Button';

import '@testing-library/jest-dom';

describe('Unit test on <Button />', () => {
    const onCLickButtonHandler = jest.fn();
    let wrapper = mount(<Button onClick={onCLickButtonHandler} textLabel="Test Btn" />)

    test('should Render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('Simulate onClick', () => {
        wrapper.find('button').simulate('click')
        expect(onCLickButtonHandler.mock.calls.length).toEqual(1);
    });
})
