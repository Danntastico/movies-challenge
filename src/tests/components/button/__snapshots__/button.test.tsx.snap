// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test on <Button /> should Render correctly 1`] = `
<button
  className="sc-bdnylx fCQhGE btn btn-primary"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
/>
`;
